### PR TITLE
Feature slim 1134 automated tests for functional exceptions encryption keys

### PR DESCRIPTION
**Pull request by bjornamkreutz**
SLIM-1134
[@SmartMetering]

### CODE REVIEW

**Comment by bjornamkreutz** [comment]
retest this please

File: cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      | DeviceType            | SMART_METER_E     |
+      | Hls3active            | false             |
+      | Hls4active            | false             |
+      | Hls5active            | true              |
+      | SecurityKeyEncryption |                   |
+    When the get actual meter reads request generating an error is received
+      | DeviceIdentification | TEST1024000000001 |
+    And a SOAP fault should have been returned
+      | Code    |                     802 |
+      | Message | INVALID_DLMS_KEY_FORMAT |
```
Comment (bvdzwet) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature]: What looks a bit odd, is that the scenario in the JIRA issue for connecting without an encryption key ends with a check for 801 INVALID_DLMS_KEY_ENCRYPTION, while the scenario above with the invalid key is described as to end with 802 INVALID_DLMS_KEY_FORMAT. The combinations in the JIRA issues seem more logical than the ones here in the feature.

File: cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+  Scenario: Replace keys on a device with incorrectly encrypted keys
```
Comment (bvdzwet) [inline cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature]: A very similar looking scenario is in SmartMeteringConfiguration feature. Wouldn't it be better to remove it from there, and add the step asserting that the keys are not changed in the database here? Now it looks like the same thing is tested twice, just with a different focus.

File: cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature (hunk 1 above)
Comment (bjornamkreutz) [reply cucumber-tests-platform-smartmetering/src/test/resources/features/functional-exceptions/FunctionalExceptionsEncryptionKeys.feature]: For this issue, I created a new US: https://smartsocietyservices.atlassian.net/browse/SLIM-1667.

**Pull request merged**